packages/@expo/cli/src/export/embed/resolveOptions.ts

```diff
@@ -128,6 +129,9 @@ export function resolveEagerOptionsAsync(
   if (!assetsDest) {
     destination ??= getTemporaryPath();
     assetsDest = path.join(destination, 'assets');
+    if (!existsSync(assetsDest)) {
+      mkdirSync(assetsDest, { recursive: true });
+    }
```

Comment:
Does it make more sense to move the creation of this folder next to the other file-persisting operations we do in `export:embed`? E.g. here: https://github.com/expo/expo/blob/63ec1439def1050cb58d9ccb7e789af08d0dbb31/packages/%40expo/cli/src/export/embed/exportEmbedAsync.ts#L148-L171

This function on its own should just resolve or calculate the options, from the provided arguments. Having it do persisting IO operations might hurt maintenance a bit.

Reply:
Maybe even inside the `persistMetroAssetsAsync` function, as that's probably used in the other exporting code paths too https://github.com/expo/expo/blob/main/packages/%40expo/cli/src/export/persistMetroAssets.ts#L26-L47

Reply:
Great idea, done!